ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.Data/Repositories/TaskRepository.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace ExadelTimeTrackingSystem.Data.Repositories
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using ExadelTimeTrackingSystem.Data.Configuration.Abstract;
+    using ExadelTimeTrackingSystem.Data.Models;
+    using ExadelTimeTrackingSystem.Data.Repositories.Abstract;
+    using MongoDB.Driver;
+
+    public class TaskRepository : MongoRepository<Models.Task>, ITaskRepository
+    {
+        public TaskRepository(IMongoDbSettings settings)
+            : base(settings)
+        {
+        }
+
+        public Task<List<Models.Task>> GetTasksOnDateAsync(DateTime date)
+        {
+            var filterBuilder = Builders<Models.Task>.Filter;
+            var filter = filterBuilder.Eq(d => d.Date, date);
```

Comment:
your field Date is of type DateTime, so that both date and time will be compared, so if task was created on 2022-05-24.01:40:10 and you will try to compare it with 2022-05-24.00:00:00 it will return false result. DateTime has a property Date which cuts time part, so use date.Date and rename date argument to dateTime. For saving task with only Date part in mapping from CreateTaskDTO to Task write ForMember for Date field and write mapping with usage of Date property of DateTime